### PR TITLE
m21 fix

### PR DESCRIPTION
**Pull request by daniel-ciocirlan**

### CODE REVIEW

File: src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala [hunk 1]
```diff
@@ -86,33 +88,33 @@ object MovieApp {
...
     import dsl._
-    implicit val directorDecoder: EntityDecoder[F, Director] = jsonOf[F, Director]
+    implicit val directorDecoder: EntityDecoder[F, Director] = jsonOf[F, Director] // most calls require Sync, but this requires Concurrent => require Async
     import cats.syntax.flatMap._
     import cats.syntax.functor._
     HttpRoutes.of[F] {
       case GET -> Root / "directors" / DirectorVar(director) =>
         if (director == Director("Zack", "Snyder"))
-          Ok(Director("Zack", "Snyder").asJson, Header("My-Custom-Header", "value"))
+          Ok(Director("Zack", "Snyder").asJson, Header.Raw(CIString("My-Custom-Header"), "value"))
```
Comment (daniel-ciocirlan) [inline src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala:100]: Caused by API change

File: src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala [hunk 2]
```diff
@@ -86,33 +88,33 @@ object MovieApp {
...
         else
           NotFound(s"No director called $director found")
       case req@POST -> Root / "directors" =>
         for {
           _ <- req.as[Director]
-          res <- Ok(`Content-Encoding`(ContentCoding.gzip))
+          res <- Ok.headers(`Content-Encoding`(ContentCoding.gzip))
```
Comment (daniel-ciocirlan) [inline src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala:106]: Caused by API change

File: src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala [hunk 3]
```diff
@@ -86,33 +88,33 @@ object MovieApp {
     }
   }
 
-  def directorRoutes[F[_] : Sync]: HttpRoutes[F] = {
+  def directorRoutes[F[_] : Async]: HttpRoutes[F] = {
```
Comment (daniel-ciocirlan) [inline src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala:91]: This is caused by the fact that most calls need a Sync instance, but jsonOf requires a Concurrent, which is unrelated to Sync.

The only way I could find to reconcile this conflict is to require an Async instance, which derives from both. This also propagates to subsequent requirements.

To be honest, I have no idea why jsonOf requires Concurrent. From what I managed to read, the minimum requirement is ApplicativeError + Monad. I only read for a few minutes so I might have missed something, but Concurrent stills seems too stringent.

File: src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala (hunk 1 above)
Comment (rcardin) [reply src/main/scala/in/rcard/http4s/tutorial/movie/MovieApp.scala:100]: Mmm...I don't think they really improve the syntax with this change. Thanks :)

**Pull request merged**